sakura_core/charset/charcode.h

```diff
@@ -27,60 +27,47 @@
 #pragma once
 
 //2007.09.13 kobake 作成
+#include <array>
 #include "parse/CWordParse.h"
 #include "util/std_macro.h"
 
-// -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- //
-//                         判定関数                            //
-// -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- //
-// #include "charset/codechecker.h"
-// SJIS関連コードは codecheker.hに移動
-
 // -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- //
 //                           定数                              //
 // -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- //
 // SJISのコードページ(CP_ACP では無くこれを使えばおそらく英語版Winでも動くはず。)	2008/5/12 Uchi
-#define CP_SJIS		932
-
-//定数の素 (直接使用は控えてください)
-#define TAB_ 				'\t'
-#define SPACE_				' '
-#define CR_					'\015'
-#define LF_					'\012'
-#define ESC_				'\x1b'
-#define CRLF_				"\015\012"
+constexpr int CP_SJIS = 932;
 
 //UNICODE定数
 namespace WCODE{
 	//文字
-	static const wchar_t TAB   = LCHAR(TAB_);
-	static const wchar_t SPACE = LCHAR(SPACE_);
-	static const wchar_t CR    = LCHAR(CR_);
-	static const wchar_t LF    = LCHAR(LF_);
-	static const wchar_t ESC   = LCHAR(ESC_);
+	constexpr wchar_t TAB   = L'\t';
+	constexpr wchar_t SPACE = L' ';
+	constexpr wchar_t CR    = L'\015';
+	constexpr wchar_t LF    = L'\012';
+	constexpr wchar_t ESC   = L'\x1b';
 
 	//文字列
-	static const wchar_t CRLF[] = LTEXT(CRLF_);
+	constexpr wchar_t CRLF[] = L"\015\012";
 
 	//特殊 (BREGEXP)
 	//$$ UNICODE版の仮デリミタ。bregonigの仕様がよくわかんないので、とりあえずこんな値にしてます。
-	static const wchar_t BREGEXP_DELIMITER = (wchar_t)0xFFFF;
+	constexpr wchar_t BREGEXP_DELIMITER = (wchar_t)0xFFFF;
 }
 
 // -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- //
 //                         判定関数                            //
 // -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- //
 
 /*! キーワードキャラクタ */
-extern const unsigned char gm_keyword_char[128];
+extern const std::array<unsigned char, 128> gm_keyword_char;
 
 //Oct. 31, 2000 JEPRO  TeX Keyword のために'\'を追加
 //Nov.  9, 2000 JEPRO  HSP Keyword のために'@'を追加
 //Oct. 18, 2007 kobake UNICODE用に書き直し
 //Nov. 27, 2010 syat   速度改善のためテーブルに変更
-inline bool IS_KEYWORD_CHAR(wchar_t wc)
+[[nodiscard]] inline bool IS_KEYWORD_CHAR(wchar_t wc)
 {
-	if(0 <= wc && wc < _countof(gm_keyword_char) && (gm_keyword_char[wc] == CK_CSYM||gm_keyword_char[wc] == CK_UDEF) )
+	if(wc < gm_keyword_char.size() && (gm_keyword_char[wc] == CK_CSYM||gm_keyword_char[wc] == CK_UDEF) )
```

Comment:
```suggestion
	if(0 <= wc && wc < gm_keyword_char.size() && (gm_keyword_char[wc] == CK_CSYM||gm_keyword_char[wc] == CK_UDEF) )
```
wchar_t は signed short だったはずなので 0 <= wc が必要だと思います。（ちょっと怪しい

Reply:
ここは警告が誤っているのではないか？と思い調べましたが、[MSVC では unsigned だそうです。](https://docs.microsoft.com/en-us/cpp/build/reference/zc-wchar-t-wchar-t-is-native-type?view=msvc-160)
> Microsoft implements wchar_t as a two-byte unsigned value.

MinGW G++ の仕様については確認しなかったので問題を作っている可能性がありますね。調べておきます。

Reply:
- size_t に変換してから比較するよう修正しました。
- 負の int を wchar_t 型にキャストして渡してみるテストケースを追加しました。